iohk/UpdateLogic.hs

```diff
@@ -77,22 +102,25 @@ fetchUrl extraHeaders url = do
   resp <- httpLbs req' man
   return $ responseBody resp
 
-fetchJson' :: FromJSON a => RequestHeaders -> String -> IO (Maybe a)
+fetchJson' :: FromJSON a => RequestHeaders -> String -> IO a
 fetchJson' extraHeaders url = do
   json <- fetchUrl extraHeaders url
   let
     parseReply :: FromJSON a => LBS.ByteString -> Maybe a
     parseReply json = decode json
     maybeObj = parseReply json
-  return maybeObj
+  case maybeObj of
+    Just v -> return v
+    Nothing -> do
+      error "unable to parse json"
 
-fetchGithubJson :: FromJSON a => String -> IO (Maybe a)
+fetchGithubJson :: FromJSON a => String -> IO a
 fetchGithubJson url = do
   githubToken <- LBS.readFile "github_token"
```

Comment:
This should be `static/github_token`, AIUI -- we already have something using that one.